R/tools.R

```diff
@@ -48,3 +48,55 @@ asBuildPath <- function(path) {
     
     return(path)
 }
+
+updateChangeLog <- function(dir = getwd(), text = "") {
+    
+    clPath <- file.path(dir, "ChangeLog")
+    if (is.null(clPath))
+        return()
+    
+    if (!file.exists(clPath))
+        file.create(clPath)
+    
+    on.exit(file.edit(clPath))
+    
+    git <- Sys.which("git")
+    stopifnot(git != "")
+    
+    gitOutput <- system("git status --porcelain", intern = TRUE)
+    if (!length(gitOutput))
+        return()
+    
+    changedFiles <- sort(substr(gitOutput, 4, nchar(gitOutput)))
+    
+    userName <- getOption("devtools.name", default = Sys.getenv("USER"))
```

Comment:
I have found Gabor Csardi's 'whoami' package excellent for this.  That said, we may not want to bring extra dependencies in.  Then again, maybe `updateChangeLog()` wants to be in its own package?